sdk/push/src/main/java/com/klaviyo/push/KlaviyoPushService.kt

```diff
@@ -34,70 +32,68 @@ class KlaviyoPushService : FirebaseMessagingService() {
          */
         fun setPushToken(pushToken: String) {
-            val properties = KlaviyoCustomerProperties()
-                .addAppendProperty(REQUEST_PUSH_KEY, pushToken)
+            val profile = Profile().addAppendProperty(PUSH_TOKEN_APPEND_KEY, pushToken)
 
-            Klaviyo.identify(properties)
+            Klaviyo.setProfile(profile)
 
-            KlaviyoPreferenceUtils.writeStringPreference(PUSH_TOKEN_PREFERENCE_KEY, pushToken)
+            Klaviyo.Registry.dataStore.store(PUSH_TOKEN_KEY, pushToken)
         }
 
         /**
-         * Returns the current push token that we have stored on this device
+         * Retrieve the device FCM push token  have stored on this device
          *
          * @return The push token we read from the shared preferences
          */
-        fun getCurrentPushToken(): String {
-            return KlaviyoPreferenceUtils.readStringPreference(PUSH_TOKEN_PREFERENCE_KEY) ?: ""
+        internal fun getPushToken(): String {
+            return Klaviyo.Registry.dataStore.fetch(PUSH_TOKEN_KEY) ?: ""
         }
 
         /**
          * Logs an $opened_push event for a remote notification that originated from Klaviyo
          *
-         * TODO customer and event properties should both be optional, once SDK is tracking customer state internally
+         * @param payload The data attributes of the push notification payload
+         */
+        internal fun openedPush(payload: Map<String, String>) {
+            payload["_k"] ?: return // Track only pushes originating from klaviyo
+
+            val event = Event().also {
+                payload.forEach { (k, v) -> it.setProperty(k, v) }
+                it.setProperty(PUSH_TOKEN_KEY, getPushToken())
+            }
+
+            Klaviyo.createEvent(KlaviyoEventType.OPENED_PUSH, event)
+        }
+
+        /**
+         * Handles a push received while app is in the foreground
          *
-         * @param pushPayload The data attributes of the push notification payload
-         * @param customerProperties Profile with which to associate the event
-         * @param eventProperties Optional additional properties for the event
+         * @param message
          */
-        fun handlePush(
-            pushPayload: Map<String, String>,
-            customerProperties: KlaviyoCustomerProperties,
-            eventProperties: KlaviyoEventProperties? = null
-        ) {
-            pushPayload["_k"] ?: return // Track pushes originating from klaviyo
-            val properties = eventProperties ?: KlaviyoEventProperties()
-            properties.addCustomProperty("push_token", getCurrentPushToken())
-            pushPayload.forEach { (k, v) -> properties.addCustomProperty(k, v) }
-            Klaviyo.track(KlaviyoEvent.OPENED_PUSH, customerProperties, properties)
+        fun handlePush(message: RemoteMessage): Boolean {
+            message.data["_k"] ?: return false
+            // TODO could we display the notification in system tray?
```

Comment:
Sorry not sure I'm following what are you referring to here with this TODO

Reply:
I'm proposing our SDK could automatically display the notification to the app user when received in the foreground. Basically, that functionality that I had to add to the _tester app_ so it could show a foreground notification, the SDK would do that for you.

Reply:
Oh I see, thats what had me confused but I had a feeling it was foreground push related. Makes sense!